specification/nginx/resource-manager/NGINX.NGINXPLUS/preview/2025-03-01-preview/swagger.json

```diff
@@ -0,0 +1,2850 @@
+          {
+            "$ref": "#/parameters/DeploymentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ConfigurationNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "The NGINX configuration",
+            "schema": {
+              "$ref": "#/definitions/NginxConfigurationRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully Created or Updated the configuration",
+            "schema": {
+              "$ref": "#/definitions/NginxConfigurationResponse"
+            }
+          },
+          "201": {
+            "description": "Configuration upsert accepted for provisioning",
+            "schema": {
+              "$ref": "#/definitions/NginxConfigurationResponse"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Configurations_CreateOrUpdate": {
+            "$ref": "./examples/Configurations_CreateOrUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "NginxConfiguration"
+        ],
+        "summary": "Reset the NGINX configuration of given NGINX deployment to default",
+        "operationId": "Configurations_Delete",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DeploymentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ConfigurationNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Configuration deletion accepted",
+            "headers": {
+              "Retry-After": {
+                "description": "Number of seconds to wait before retrying or until the operation is likely to complete",
+                "type": "integer",
+                "format": "int32"
+              },
+              "Operation-Location": {
```

Comment:
>"Operation-Location": { [](http://example.com/codeflow?start=14&length=23)

why Operation-Location? Most Azure APIs use the Azure-AsyncOperation header for this I think.

[https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/async-operations](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/async-operations)

Reply:
Changed to Azure-AsyncOperation